test/unit/iso_generator_test.rb

```diff
@@ -3,46 +3,36 @@
 class ForemanBootdisk::IsoGeneratorTest < ActiveSupport::TestCase
   include ForemanBootdiskTestHelper
   setup :setup_bootdisk
-  setup :setup_org_loc
-  setup :setup_subnet
-  setup :setup_host
 
-  setup do
-    @host.build = true
-  end
+  describe '#generate_full_host' do
+    let(:medium) { FactoryBot.create(:medium, name: 'Red Hat Enterprise Linux Atomic Mirror') }
+    let(:operatingsystem) { FactoryBot.create(:ubuntu14_10, :with_archs, :with_ptables, media: [medium]) }
+    let(:host) { FactoryBot.create(:host, :managed, operatingsystem: operatingsystem, build: true) }
+    let(:template) { FactoryBot.create(:provisioning_template, template: 'Fake kernel line <%= @kernel %> - <%= @initrd %>') }
 
-  test "generate_full_host creates with ISO-compatible file names" do
-    @host.expects(:generate_pxe_template).with(:PXELinux).returns("Fake kernel line boot/Debian-8.1-x86_64-vmlinuz")
-    boot_files = [
-      {:"boot/Debian-8.1-x86_64"=>"/tmp/pxeboot/vmlinuz"},
-      {:"boot/Debian-8.1-x86_64"=>"/tmp/pxeboot/initrd.img"}
-    ]
-    @host.operatingsystem.expects(:pxe_files).returns(boot_files)
-    ForemanBootdisk::ISOGenerator.expects(:generate).with({
-      :isolinux => 'Fake kernel line BOOT/DEBIAN_8_1_X86_64_VMLINUZ',
-      :files => [[['BOOT/DEBIAN_8_1_X86_64_VMLINUZ', '/tmp/pxeboot/vmlinuz']],
-      [['BOOT/DEBIAN_8_1_X86_64_INITRD_IMG', '/tmp/pxeboot/initrd.img']]] }, anything)
-    ForemanBootdisk::ISOGenerator.generate_full_host(@host)
-  end
+    setup do
+      host.expects(:provisioning_template).with(kind: :PXELinux).returns(template)
+    end
 
-  test "generate_full_host creates with ISO-compatible long file names" do
-    @host.expects(:generate_pxe_template).with(:PXELinux).returns("Fake kernel line boot/RedHatEnterpriseLinuxAtomic-7.3-x86_64-vmlinuz")
-    boot_files = [
-      {:"boot/RedHatEnterpriseLinuxAtomic-7.3-x86_64"=>"/tmp/pxeboot/vmlinuz"},
-      {:"boot/RedHatEnterpriseLinuxAtomic-7.3-x86_64"=>"/tmp/pxeboot/initrd.img"}
-    ]
-    @host.operatingsystem.expects(:pxe_files).returns(boot_files)
-    ForemanBootdisk::ISOGenerator.expects(:generate).with({
-      :isolinux => 'Fake kernel line BOOT/NUXATOMIC_7_3_X86_64_VMLINUZ',
-      :files => [[['BOOT/NUXATOMIC_7_3_X86_64_VMLINUZ', '/tmp/pxeboot/vmlinuz']],
-        [['BOOT/ATOMIC_7_3_X86_64_INITRD_IMG', '/tmp/pxeboot/initrd.img']]]}, anything)
-    ForemanBootdisk::ISOGenerator.generate_full_host(@host)
-  end
```

Comment:
This test is actually for `iso9660_filename` to make sure that filenames gets shortened to maximum size for ISO format, can you keep it in some way or another? Note `boot/RedHatEnterpriseLinuxAtomic-7.3-x86_64` -> `BOOT/NUXATOMIC_7_3_X86_64_VMLINUZ` (shortened automatically).

Reply:
Sure, I can add some tests for `iso9660_filename` method. We should definitely have them.

Reply:
Yep should be probably a unit test, whatever you prefer.